src/client/replication_ddl_client.cpp

```diff
@@ -1547,5 +1549,36 @@ void replication_ddl_client::query_disk_info(
     call_rpcs_async(query_disk_info_rpcs, resps);
 }
 
+error_with<start_bulk_load_response>
+replication_ddl_client::start_bulk_load(const std::string &app_name,
+                                        const std::string &cluster_name,
+                                        const std::string &file_provider_type)
+{
+    auto req = make_unique<start_bulk_load_request>();
+    req->app_name = app_name;
+    req->cluster_name = cluster_name;
+    req->file_provider_type = file_provider_type;
+    return call_rpc_sync(start_bulk_load_rpc(std::move(req), RPC_CM_START_BULK_LOAD));
+}
+
+error_with<control_bulk_load_response>
+replication_ddl_client::control_bulk_load(const std::string &app_name,
+                                          bulk_load_control_type::type control_type)
+{
+    auto req = make_unique<control_bulk_load_request>();
+    req->app_name = app_name;
+    req->type = control_type;
+    return call_rpc_sync(control_bulk_load_rpc(std::move(req), RPC_CM_CONTROL_BULK_LOAD));
+}
+
+error_with<query_bulk_load_response>
+replication_ddl_client::query_bulk_load(const std::string &app_name)
+{
+
+    auto req = make_unique<query_bulk_load_request>();
+    req->app_name = app_name;
```

Comment:
`__set_app_name`

Reply:
I think `app_name` is a required filed, it is okay to set its value directly.

Reply:
`__set_app_name` will always right, but it is wrong to use `app_name` directly when someone change it to optional.

Reply:
If you would like to update a required filed into optional, you should also update the related code. In my view, `__set_app_name` is used for optional filed, the programmer who reads the code will figure out which filed is optional or required easily.